### ISSUE TITLE
dirbuf isn't used if I specify a directory from the command line

### ISSUE THREAD

**Issue by andrewferrier** [elihunter173/dirbuf.nvim#36]
If I open NeoVim directly with a directory, like this: `nvim .`, then `dirbuf.nvim` is not used for that buffer, instead it opens in `netrw`. Subsequent directory-based buffers use `dirbuf` (for example, if I then type `:edit ~`, that opens in `dirbuf`), but the initial buffer does not. If I just open NeoVim with `nvim`, then `:edit ~`, that also opens in `dirbuf` as expected.

I thought this used to work (with `dirbuf`), but that could be my imagination.

I am loading `dirbuf` in a conventional way using `packer.nvim` (essentially, it ends up in `~/.local/share/nvim/site/pack/packer/start/dirbuf.nvim`). Relevant packer section looks like:

```
    use({
        "elihunter173/dirbuf.nvim",
        config = function()
            require("dirbuf").setup({
                 hash_padding = 5,
                 sort_order = "directories_first",
            })
        end,
    })
```

I've tried looking at [this section of the code](https://github.com/elihunter173/dirbuf.nvim/blob/ebbcc7d686dc3c87778eee5630d85bf9d995b2fc/plugin/dirbuf.vim#L29), and I can see that dirbuf attempts to remove netrw's `FileExplorer`. I wonder if that's not being loaded soon enough?

**Comment by andrewferrier** [comment]
Incidentally, if it helps, I tried a workaround of adding `vim.g.loaded_netrw = true` to my `init.lua`.

When I do this, and open a directory from the arguments, I now get an error like this:

```
"." is a directory
Error detected while processing function <SNR>21_VimEnter[10]..<SNR>21_LocalBrowse:
line   32:
E117: Unknown function: netrw#LocalBrowseCheck
```

**Comment by andrewferrier** [comment]
NVIM v0.7.0 on MacOS, FWIW.

**Comment by andrewferrier** [comment]
OK, you can close this issue. I just read the notes here: https://github.com/elihunter173/dirbuf.nvim#notes. Realized I should be setting `vim.g.loaded_netrwPlugin = 1` also.

Sorry :( Should have RTFM.

(Devil's advocate: could/should `dirbuf` be setting these for me?)

**Comment by elihunter173** [comment]
I'll leave this issue open until I land on a decision on what if any changes to make to make this easier/cleaner.

>(Devil's advocate: could/should `dirbuf` be setting these for me?)

I have mixed feelings about this. In my previous investigation, you have to set-up the Netrw-hijacking autocommand before Netrw loads, so the most reliable place to do this would be in `dirbuf.setup()`. But I don't like `dirbuf.setup()` doing more than just setting variables because it means you *have* to call setup for full functionality and it makes it difficult to keep the setup function idempotent.

I'll first see if there's anything I can do in `plugin/dirbuf.vim` to make this work so calling `dirbuf.setup()` remains completely optional. If that doesn't work, I might add something to setup to make it work, but I'm not sure yet.

> Sorry :( Should have RTFM.

No need to apologize! Dirbuf.nvim tries to be as intuitive as possible and reading a manual isn't conducive to this ;). And for the record I don't really like the current warning in the README either

**Comment by andrewferrier** [comment]
Yep, all makes sense, thanks. I don't have a problem requiring `setup()` to be called incidentally; many other NeoVim plugins do the same. But I never attempt to reload my NeoVim config; I quit and restart. Maybe I'm in the minority.

I would have thought `setup()` could look something like this by the way:

```
setup()
    if not vim.g.dirbuf_loaded
           // Hijack Netrw autocmds
    end
    vim.g.dirbuf_loaded = true 
end
```

**Comment by andrewferrier** [comment]
Sorry for the delay. Confirmed that the `netrw-issue` branch seems to fix the issue. Even if I don't include a `setup()` call in my config, or `vim.g.loaded_netrwPlugin = 1`, opening NeoVim straight into a directory opens it in `dirbuf`, which is fantastic. Switching back to `main` removes that behaviour. I think you ought to be able to merge this.

**Comment by elihunter173** [comment]
Fixed by d610602ebb81a071bef1674c4b0b5fa98f111204. I made a typo in the commit message without catching it, so the issue didn't auto-close :'(
